syntax = "proto3";

service DbnodeService {
    rpc GetDocument (GetDocumentRequest) returns (DocumentResponse) {}
    rpc UpdateDocument (UpdateDocumentRequest) returns (Empty) {}
    rpc HeartBeat (HeartBeatRequest) returns (Empty);
    rpc Election (RunElectionRequest) returns (Empty);
}

message GetDocumentRequest {
    string collection_name = 1;
    string document_id = 2;
}

message DocumentResponse {
    string document = 1;
}

message UpdateDocumentRequest {
    string collection_name = 1;
    string document = 2;
}

message HeartBeatRequest {
    uint32 process_id = 1;
}

message RunElectionRequest {
    uint32 process_id = 1;
}

message Empty {}